### PR TITLE
chore: autorouter migrate to v6

### PR DESCRIPTION
**Pull request by PeeyushDas**
### Your checklist for this pull request

🚨Please review the [guidelines for contributing](https://github.com/Monday-Morning/project-pegasus/blob/main/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your default branch!
- [x] Make sure you are making a pull request against the **default branch** (left side). Also you should start _your branch_ off _default branch_.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.
- [x] I have added necessary documentation (if appropriate)

### Linked Issues

Fixes

### Description

Please describe your pull request.

❤️Thank you!

### Post merge checklist

- [x] Follow steps from the [guidelines for contributing](https://github.com/Monday-Morning/project-pegasus/blob/main/CONTRIBUTING.md) to this repository.
- [x] If you are a new contributor, ping in the thread and one of the maintainers will add you to all-contributors list.

### CODE REVIEW

File: app/android/app/build.gradle [hunk 1]
```diff
@@ -82,7 +82,7 @@ android {
 
 		signingConfigs {
 				release {
-						storeFile file(keystoreProperties['storeFile'])
+						storeFile file("key.jks")
```
Comment (rutajdash) [inline app/android/app/build.gradle:85]: Why was this change committed?

File: app/lib/src/app.dart [hunk 2]
```diff
@@ -4,9 +4,9 @@ import 'package:flutter/services.dart';
 import 'package:hooks_riverpod/hooks_riverpod.dart';
 import 'package:mondaymorning/src/pages/full_error_page.dart';
 import 'package:mondaymorning/src/pages/splash_page.dart';
-import 'package:mondaymorning/src/services/router/mm_router.dart';
 import 'package:mondaymorning/src/services/themes/index.dart';
 import 'package:mondaymorning/src/store/states/app_config/app_config_provider.dart';
+import 'services/router/mm_router.dart';
```
Comment (rutajdash) [inline app/lib/src/app.dart:9]:
```suggestion
import 'package:mondaymorning/src/services/router/mm_router.dart';
```
Avoid using relative imports.

File: app/lib/src/pages/article_page.dart [hunk 3]
```diff
@@ -8,8 +8,8 @@ import 'package:mondaymorning/src/store/states/article_page/article_page_provide
 import 'package:mondaymorning/src/ui/screens/article_screen.dart';
 import 'package:mondaymorning/src/ui/screens/full_article_loading_screen.dart';
 import 'package:mondaymorning/src/ui/screens/full_error_screen.dart';
-import 'package:mondaymorning/src/ui/screens/full_loading_screen.dart';
 
+@RoutePage<dynamic>(name: 'ArticleRoute')
```
Comment (rutajdash) [inline app/lib/src/pages/article_page.dart:12]: Why dynamic? This page doesn't return any data so it should be void instead.


```suggestion
@RoutePage<void>(name: 'ArticleRoute')
```

File: app/lib/src/pages/full_error_page.dart [hunk 4]
```diff
@@ -1,7 +1,9 @@
+import 'package:auto_route/auto_route.dart';
 import 'package:flutter/material.dart';
 import 'package:mondaymorning/src/services/themes/index.dart';
 import 'package:mondaymorning/src/ui/screens/full_error_screen.dart';
 
+@RoutePage<dynamic>(name: 'NotFoundRoute')
```
Comment (rutajdash) [inline app/lib/src/pages/full_error_page.dart:6]: Why dynamic? This page doesn't return any data so it should be void instead.

```suggestion
@RoutePage<void>(name: 'NotFoundRoute')
```

File: app/lib/src/pages/landing_page.dart [hunk 5]
```diff
@@ -2,14 +2,15 @@ import 'package:auto_route/auto_route.dart';
...
-import 'package:mondaymorning/src/services/themes/size_config.dart';
 import 'package:mondaymorning/src/services/router/mm_router.dart';
+import 'package:mondaymorning/src/services/themes/size_config.dart';
 import 'package:mondaymorning/src/store/states/landing_page/landing_page_data_type.dart';
 import 'package:mondaymorning/src/store/states/landing_page/landing_page_provider.dart';
 import 'package:mondaymorning/src/ui/components/home/landing_bottom_bar.dart';
 import 'package:mondaymorning/src/ui/screens/full_error_screen.dart';
 import 'package:mondaymorning/src/ui/screens/full_loading_screen.dart';
 
+@RoutePage<dynamic>(name: 'LandingRoute')
```
Comment (rutajdash) [inline app/lib/src/pages/landing_page.dart:13]: Why dynamic? This page doesn't return any data so it should be void instead.

```suggestion
@RoutePage<void>(name: 'LandingRoute')
```

File: app/lib/src/ui/components/terms/content_screen.dart [hunk 6]
```diff
@@ -1,7 +1,9 @@
+import 'package:auto_route/auto_route.dart';
 import 'package:flutter/material.dart';
 import 'package:mondaymorning/src/services/themes/mm_colors.dart';
 import 'package:mondaymorning/src/ui/components/terms/terms_content.dart';
 
+@RoutePage<dynamic>(name: 'ContentRoute')
```
Comment (rutajdash) [inline app/lib/src/ui/components/terms/content_screen.dart:6]: Why dynamic? This page doesn't return any data so it should be void instead.


```suggestion
@RoutePage<void>(name: 'ContentRoute')
```

File: app/lib/src/ui/screens/about_screen.dart [hunk 7]
```diff
@@ -1,10 +1,12 @@
+import 'package:auto_route/auto_route.dart';
 import 'package:flutter/material.dart';
 import 'package:mondaymorning/src/services/themes/size_config.dart';
 import 'package:mondaymorning/src/ui/components/more/about/cc_card.dart';
 import 'package:mondaymorning/src/ui/components/more/about/email_card.dart';
 import 'package:mondaymorning/src/ui/components/more/about/map_card.dart';
 import 'package:mondaymorning/src/ui/components/more/about/social_card.dart';
 
+@RoutePage(name: 'AboutRoute')
```
Comment (rutajdash) [inline app/lib/src/ui/screens/about_screen.dart:9]: Maintain consistency

```suggestion
@RoutePage<void>(name: 'AboutRoute')
```

File: app/lib/src/ui/screens/categories_screen.dart [hunk 8]
```diff
@@ -1,10 +1,12 @@
+import 'package:auto_route/auto_route.dart';
 import 'package:flutter/material.dart';
 import 'package:mondaymorning/src/api/models/squiggle/squiggle.dart';
 import 'package:mondaymorning/src/services/themes/size_config.dart';
 import 'package:mondaymorning/src/ui/components/categories/categories_app_bar.dart';
 import 'package:mondaymorning/src/ui/components/categories/main_categories_grid.dart';
 import 'package:mondaymorning/src/ui/components/categories/squiggle_box.dart';
 
+@RoutePage<dynamic>(name: 'CategoriesRoute')
```
Comment (rutajdash) [inline app/lib/src/ui/screens/categories_screen.dart:9]:
```suggestion
@RoutePage<void>(name: 'CategoriesRoute')
```

File: app/lib/src/ui/screens/category_screen.dart [hunk 9]
```diff
@@ -6,6 +7,7 @@ import 'package:mondaymorning/src/store/constants/categories.dart';
 import 'package:mondaymorning/src/ui/components/article/article_carousel.dart';
 import 'package:mondaymorning/src/ui/components/category/sub_category_section.dart';
 
+@RoutePage<dynamic>(name: 'AllCategoryRoute')
```
Comment (rutajdash) [inline app/lib/src/ui/screens/category_screen.dart:10]:
```suggestion
@RoutePage<void>(name: 'AllCategoryRoute')
```

File: app/lib/src/ui/screens/home_screen.dart [hunk 10]
```diff
@@ -1,13 +1,14 @@
...
 import 'package:mondaymorning/src/api/models/article/article.dart';
+import 'package:mondaymorning/src/services/router/mm_router.dart';
 import 'package:mondaymorning/src/services/themes/size_config.dart';
 import 'package:mondaymorning/src/store/states/landing_page/landing_page_data_type.dart';
 import 'package:mondaymorning/src/ui/components/article/article_carousel.dart';
 import 'package:mondaymorning/src/ui/components/article/small_article_card.dart';
 import 'package:mondaymorning/src/ui/components/home/search_app_bar.dart';
-import 'package:mondaymorning/src/services/router/mm_router.dart';
 
+@RoutePage<dynamic>(name: 'HomeRoute')
```
Comment (rutajdash) [inline app/lib/src/ui/screens/home_screen.dart:11]:
```suggestion
@RoutePage<void>(name: 'HomeRoute')
```

File: app/lib/src/ui/screens/more_screen.dart [hunk 11]
```diff
@@ -19,16 +16,17 @@ class MoreCard {
 }
 
 /// More Screen of the app.
+@RoutePage<dynamic>(name: 'MoreRoute')
```
Comment (rutajdash) [inline app/lib/src/ui/screens/more_screen.dart:19]:
```suggestion
@RoutePage<void>(name: 'MoreRoute')
```

File: app/lib/src/ui/screens/sub_category_screen.dart [hunk 12]
```diff
@@ -7,6 +7,7 @@ import 'package:mondaymorning/src/services/themes/size_config.dart';
 import 'package:mondaymorning/src/ui/components/article/article_carousel.dart';
 import 'package:mondaymorning/src/ui/components/article/small_article_card.dart';
 
+@RoutePage<dynamic>(name: 'SubCategoryRoute')
```
Comment (rutajdash) [inline app/lib/src/ui/screens/sub_category_screen.dart:10]:
```suggestion
@RoutePage<void>(name: 'SubCategoryRoute')
```

File: app/lib/src/ui/screens/terms_and_policy_screen.dart [hunk 13]
```diff
@@ -1,9 +1,10 @@
 import 'package:auto_route/auto_route.dart';
 import 'package:flutter/material.dart';
+import 'package:mondaymorning/src/services/router/mm_router.dart';
 import 'package:mondaymorning/src/ui/components/terms/terms_app_bar.dart';
 import 'package:mondaymorning/src/ui/components/terms/terms_content.dart';
-import 'package:mondaymorning/src/services/router/mm_router.dart';
 
+@RoutePage<dynamic>(name: 'TermsRoute')
```
Comment (rutajdash) [inline app/lib/src/ui/screens/terms_and_policy_screen.dart:7]:
```suggestion
@RoutePage<void>(name: 'TermsRoute')
```

**Review by rutajdash — CHANGES_REQUESTED** [review]
1. Are we migrating to Material 3? If yes, why? If not, why was the design system changed?

2. Why is the coming soon page renamed to the expression page?

File: app/android/app/build.gradle (hunk 1 above)
Comment (PeeyushDas) [reply app/android/app/build.gradle:85]: @rutajdash 
Sir, when i try to run the app with 
storeFile file(keystoreProperties['storeFile'])

The following error pops up in the terminal
Screenshot from 2023-09-20 23-18-46
Launching lib/main.dart on SM E225F in debug mode...

FAILURE: Build failed with an exception.

* Where:
Build file '/home/peeyushdas/Desktop/MONDAY MORNING 2023/project-pegasus/app/android/app/build.gradle' line: 85

* What went wrong:
A problem occurred evaluating project ':app'.
> path may not be null or empty string. path='null'


And then I searched for the issue on stackoverflow, and I got this solution:-

link:- https://stackoverflow.com/questions/54457245/a-problem-occurred-evaluating-project-app-path-may-not-be-null-or-empty-st

Screenshot from 2023-09-20 23-18-15

Comment (rutajdash) [reply app/android/app/build.gradle:85]: Yes, we use a keystore properties file for production builds along with credential files, which for obvious reasons is not shared to all developers.
For development, making this change is acceptable but it should not be committed but only kept on your local system.

Comment (PeeyushDas) [reply app/android/app/build.gradle:85]: Ok sir.

**Comment by sambit-sankalp** [comment]
We need to make another PR as there is a change in the visibility of the repository.
Link to other PR - https://github.com/Monday-Morning/project-pegasus/pull/74

We can close this PR and start reviewing the other,.

**Pull request closed without merging**